### ISSUE TITLE
How to make it work faster?

### ISSUE THREAD

**Issue by masihyeganeh** [hpgrahsl/kafka-connect-mongodb#60]
Hi.
I need to thank you for this project. It works well in our company.
But I need to know if there is a way to make it transfer data faster or not.
Here is a screenshot of our Control Center:
<img width="1294" alt="screen shot 1397-08-20 at 5 39 12 pm" src="https://user-images.githubusercontent.com/692335/48314032-f7ecfb80-e5d8-11e8-8c1f-f5c06ca049d5.png">

As you can see, it is over 9 million records behind producer.
So, how can I make it work faster?
Maybe having multiple sink that work as a consumer group? I don't know.
Your advice can really help us here.

Regards

**Comment by hpgrahsl** [comment]
Hi @masihyeganeh !

I'm happy to hear that the connector is useful for you/your company :) Your question is extremely general and also the screenshot showing the lag of roughly ~9Mio messages doesn't help much unfortunately.

1) What is the average ingestion rate to the topic that the sink connector consumes i.e. how many messages / sec are produced to this topic? During which period (a few minutes, hours, days,...) of processing did you fall behind by this number of records?

2) How many partitions are configured for the topic in question?

3) I assume you are running connect in distributed mode. How many worker nodes do you run?

4) How does your sink connector configuration look like? I hope you don't run just 1 task but at least a few.

These are just the most relevant questions which you should try to answer first in order do understand why the sink connector is running behind.

Rest assured that by leveraging the correct configuration for connect itself and the connector instance in order to allow for proper parallelism during the consumption you should be able to drastically reduce your lag and come very close to near real-time.

Finally you should also make sure that your MongoDB setup can deal with the write load that will result from this. If you operate e.g. against a ReplicaSet you have to check if your primary node can manage the write load.

The bottom line of this is that while the sink connector code could definitely tuned and improved performance-wise I haven't heard yet that the connector itself was the bottleneck for the use cases I know about.

Talking about your use case again I would highly appreciate if you would provide me a brief description which I can use as a user voice / customer testimonial in the README :)

**Comment by hpgrahsl** [comment]
Hi @masihyeganeh, since I haven't heard back for > 3 weeks I think it makes most sense to close this issue for now. In case you find the time to clarify in detail and answer at least some of my questions above we can dig deeper and reopen. Until then all the best!
